docs/networking/rancher-desktop-guest-agent.md

```diff
@@ -0,0 +1,149 @@
+
+-   **containerdSock**: File path for the containerd socket address. If no argument is provided, it defaults to `/run/k3s/containerd/containerd.sock`.
+
+-   **vtunnelAddr**: Peer address for the Vtunnel process that forwards port mappings to the Vtunnel Host process over `AF_VSOCK`. This feature will soon be deprecated.
+
+-   **privilegedService**: This flag enables privileged service mode. When enabled, the Rancher Desktop guest agent communicates over `AF_VSOCK` to a privileged service process running on the host. This functionality is primarily designed for Windows and will soon be deprecated.
+
+-   **k8sServiceListenerAddr**: Specifies an IP address (`0.0.0.0` or `127.0.0.1`) to bind Kubernetes services on the host.
+
+-   **adminInstall**: This flag indicates whether Rancher Desktop is installed with administrator privileges. It is used to enable Network Tunnel mode, where port mappings are forwarded to Rancher Desktop Networking's `host-switch`. The `host-switch` hosts an API that exposes ports from the host into the network namespace.
+
+-   **k8sAPIPort**: Specifies the Kubernetes API port, which is forwarded to `wsl-proxy` to allow other distros that are part of WSL integrations to  interact via `kubectl`.
+
+## PortMapping
+
+Is a struct object that represents an exposed container or a service. [Portmapping](https://github.com/rancher-sandbox/rancher-desktop-agent/blob/8348a5e10578f8662532eddfac73b243cfd419f4/pkg/types/portmapping.go#L22) objects consist of the following fields:
+
+```
+type PortMapping struct {
+	// Remove indicates whether to remove or add the entry
+	Remove bool `json:"remove"`
+	// Ports are the port mappings for both IPV4 and IPV6
+	Ports nat.PortMap `json:"ports"`
+	// ConnectAddrs are the backend addresses to connect to
+	ConnectAddrs []ConnectAddrs `json:"connectAddrs"`
+}
+```
+## Networking Mode
+
+The Rancher Desktop Guest Agent currently supports two networking modes. In the first mode, if the privileged service flag mentioned above is enabled, it utilizes the vtunnel peer process to forward all port mappings to privileged services running on the host, thereby creating port proxies. However, this mode lacks network namespace isolation and shares iptables with other WSL distros, and it will soon be deprecated.
```

Comment:
… But the third paragraph says there are three modes.

Perhaps it would be easier to understand as a list:

Rancher Desktop Guest Agent can operate in one of three networking modes, depending on startup arguments:

<dl>
<dt><tt>-privilegedService</tt>
<dd>Rancher Desktop utilizes the vtunnel peer process to forward all port mappings to privileged services running on the host, thereby creating port proxies. However, this mode lacks network namespace isolation and shares iptables with other WSL distros, and it will soon be deprecated.
<dt><tt>-adminInstall</tt>
<dd>The Network Tunnel mode allows the Guest Agent to operate in an isolated network namespace with a dedicated iptables. This mode is enabled through Rancher Desktop Networking.
<dt>Neither <tt>-privilegedService</tt> nor <tt>-adminInstall</tt>
<dd>Rancher Desktop Guest Agent operates in non-admin user mode. In this mode, all port mappings are bound to localhost, and the use of privileged ports is restricted.
</dl>